src/systemdhook.c

```diff
@@ -58,6 +68,113 @@ DEFINE_CLEANUP_FUNC(yajl_val, yajl_tree_free)
 
 #define CGROUP_ROOT "/sys/fs/cgroup"
 
+static int makepath(char *dir, mode_t mode)
+{
+    if (!dir) {
+	errno = EINVAL;
+	return -1;
+    }
+
+    if (strlen(dir) == 1 && dir[0] == '/')
+	return 0;
+
+    makepath(dirname(strdupa(dir)), mode);
+
+    return mkdir(dir, mode);
+}
+
+static int bind_mount(const char *src, const char *dest, const char *opt) {
+	if (mount(src, dest, "bind", MS_BIND|MS_REC, NULL) == -1) {
+		pr_perror("Failed to mount %s on %s", src, dest);
+		return -1;
+	}
+	if (mount(src, dest, "bind", MS_REMOUNT|MS_BIND|MS_REC, opt) == -1) {
```

Comment:
Also, we could do this remount only when read-only is required and probably use MS_RDONLY instead of "ro" as option.